lib/code_corps/stripe_service/stripe_connect_external_account_service.ex

```diff
@@ -1,16 +1,24 @@
 defmodule CodeCorps.StripeService.StripeConnectExternalAccountService do
-  alias CodeCorps.{Repo, StripeExternalAccount}
+  alias CodeCorps.{Repo, StripeConnectAccount, StripeExternalAccount}
   alias CodeCorps.StripeService.Adapters.StripeExternalAccountAdapter
 
   @api Application.get_env(:code_corps, :stripe)
 
   def create(id_from_stripe, account_id_from_stripe) do
-    with {:ok, %Stripe.ExternalAccount{} = bank_account} <- @api.ExternalAccount.retrieve(id_from_stripe, connect_account: account_id_from_stripe),
-         {:ok, params} <- StripeExternalAccountAdapter.to_params(bank_account)
+    with {:ok, %Stripe.ExternalAccount{} = external_account} <- @api.ExternalAccount.retrieve(id_from_stripe, connect_account: account_id_from_stripe),
+         {:ok, %StripeConnectAccount{} = connect_account} <- get_connect_account(account_id_from_stripe),
+         {:ok, params} <- StripeExternalAccountAdapter.to_params(external_account, connect_account.id)
     do
       %StripeExternalAccount{}
       |> StripeExternalAccount.changeset(params)
       |> Repo.insert
     end
   end
+
+  defp get_connect_account(account_id_from_stripe) do
```

Comment:
The reason we do this at the service instead of the adapter level is because we want the service to return a useful error response in case the connect account was not found.

Specifically, the service will return `{:ok, created_external_account}`  if everything is ok, and `{:error, :not_found}` if the associated connect account was not found locally. 

Due to this behavior, the event handling system will mark the event as errored.